src/main/java/bio/terra/stairway/FlightMap.java

```diff
@@ -61,6 +66,57 @@ public FlightMap() {
     return inputList;
   }
 
+  /**
+   * Depending on what version of code a Flight log was written with, we may have a JSON Map, a
+   * {@code List<FlightInput>}, both, or neither at deserialization time. This method is used to
+   * generate an {@code Optional<FlightMap>} based on what was contained in the database. This will
+   * return an empty {@code Optional<FlightMap>} if json is null.
+   *
+   * @param inputList Entries in flightworking for a given Flight log. May be empty if Flight
+   *     pre-existed flightworking table, or there are no working parameters.
+   * @param json Map of working entries for a given Flight log in JSON. May be NULL.
+   */
+  static Optional<FlightMap> create(List<FlightInput> inputList, @Nullable String json) {
+
+    // Note that currently if this is NULL, it indicates that the output_parameters field of the
+    // flight table was empty.  Going forward (PF-703) we will expect NULL json to be the normal
+    // case and should also look at flightInput.
+    if (json == null) {
+      return Optional.empty();
+    }
+
+    // TODO(PF-703): Once we stop writing JSON, we may still have JSON-only flight data in the
+    //               database, as well as flights with both. At that point we should favor
+    //               inputList over json.
+
+    FlightMap map = new FlightMap();
+    map.fromJson(json);
+
```

Comment:
If we got a non-empty `inputList` from `flightworking`, the block below will validate that all of the values in `inputList` exist in the deserialized-from-JSON `FlightMap` and can be deserialized as the same type.  It does not fail if either check fails, but logs ERRORs.  Am looking for feedback on whether folks are OK with this as a way to build confidence before the next phase (where we stop writing JSON map to `flightlog::working_parameters`).